ldap_auth_provider.py

```diff
@@ -130,6 +128,10 @@ def check_auth(self, username, login_type, login_dict):
             )
 
             if self.ldap_mode == LDAPMode.SIMPLE:
+                if not password:
+                    # Password is mandatory in simple bind
+                    defer.returnValue(False)
```

Comment:
I'm still a bit surprised by this. why do we need to special-case it? why is an empty password any different to any other incorrect password?

Reply:
That's how we could introduce security issue but thanks to ldap3 we don't.  
Long story short LDAP can not authenticate user without password. Attempting to bind user without password leads to anonymous bind which always success.  
Fortunately in ldap3 anonymous bind was splitted off to special case.  

I will revert empty password changes.